File: src/class037/C6_TrimBinarySearchTree.py
# 修剪搜索二叉树
# 测试链接 : https://leetcode.cn/problems/trim-a-binary-search-tree/

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def trimBST(cur, low, high):
    if cur is None:
        return None
    if cur.val < low:
        return trimBST(cur.right, low, high)
    if cur.val > high:
        return trimBST(cur.left, low, high)
    # cur在范围中
    cur.left = trimBST(cur.left, low, high)
    cur.right = trimBST(cur.right, low, high)
    return cur

File: src/class037/C6_TrimBinarySearchTree.cpp
#include <iostream>

using namespace std;

// 修剪搜索二叉树
// 测试链接 : https://leetcode.cn/problems/trim-a-binary-search-tree/

// 不提交这个类
class TreeNode {
public:
    int val;
    TreeNode* left;
    TreeNode* right;

    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

// 提交以下的方法
// [low, high]
TreeNode* trimBST(TreeNode* cur, int low, int high) {
    if (cur == nullptr) {
        return nullptr;
    }
    if (cur->val < low) {
        return trimBST(cur->right, low, high);
    }
    if (cur->val > high) {
        return trimBST(cur->left, low, high);
    }
    // cur在范围中
    cur->left = trimBST(cur->left, low, high);
    cur->right = trimBST(cur->right, low, high);
    return cur;
}
